apps/src/templates/progress/ProgressBubble.jsx

```diff
@@ -35,272 +34,78 @@ class ProgressBubble extends React.Component {
-  };
-
-  recordProgressTabProgressBubbleClick = () => {
-    firehoseClient.putRecord(
-      {
-        study: 'teacher_dashboard_actions',
-        study_group: 'progress',
-        event: 'go_to_level',
-        data_json: JSON.stringify({
-          student_id: this.props.selectedStudentId,
-          level_url: this.props.level.url,
-          section_id: this.props.selectedSectionId
-        })
-      },
-      {includeUserId: true}
-    );
-  };
-
-  render() {
-    const {
-      level,
-      disabled,
-      smallBubble,
-      selectedSectionId,
-      selectedStudentId,
-      currentLocation,
-      lessonTrophyEnabled,
-      pairingIconEnabled,
-      hideAssessmentIcon,
-      onClick
-    } = this.props;
-
-    const levelIsAssessment = isLevelAssessment(level);
-
-    const number = level.levelNumber;
-    const url = level.url;
-    const levelName = level.name || level.progressionDisplayName;
-    const levelIcon = getIconForLevel(level);
-
-    const hideNumber =
-      level.letter || level.isLocked || level.paired || level.bonus;
-
-    const style = {
-      ...styles.main,
-      ...(!disabled && hoverStyle),
-      ...(smallBubble && styles.small),
-      ...(level.isConceptLevel &&
-        (smallBubble ? styles.smallDiamond : styles.largeDiamond)),
-      ...levelProgressStyle(level.status, level.kind),
-      ...(level.highlighted && styles.highlighted)
-    };
-
-    let href = '';
-    if (!disabled && url && !onClick) {
-      const queryParams = queryString.parse(currentLocation.search);
+  isEnabled() {
+    const {disabled, level, onClick} = this.props;
+    return !disabled && (!!level.url || !!onClick);
+  }
 
-      if (selectedSectionId) {
-        queryParams.section_id = selectedSectionId;
-      }
-      if (selectedStudentId) {
-        queryParams.user_id = selectedStudentId;
-      }
-      const paramString = queryString.stringify(queryParams);
-      href = url;
-      if (paramString.length > 0) {
-        // If href already has 1 or more query params, our delimiter will be '&'.
-        // If href has no query params, our delimiter is '?'.
-        // TODO: (madelynkasula) Refactor this logic to use queryString.parseUrl(href)
-        // instead. Our current version of query-string (4.1.0) does not yet have this method.
-        const delimiter = /\?/.test(href) ? '&' : '?';
-        href += delimiter + paramString;
-      }
+  onClickLevel = () => {
+    if (this.props.onClick) {
+      this.props.onClick(this.props.level);
     }
+  };
 
-    const tooltipId = _.uniqueId();
-    let tooltipText = level.isSublevel
-      ? level.display_name
-      : levelName || (level.isUnplugged && i18n.unpluggedActivity()) || '';
-    if (number) {
-      tooltipText = `${number}. ${tooltipText}`;
-    }
+  /**
+   * onClick takes priority over url, so if we have onClick, return null
+   */
+  getUrl() {
+    const {onClick, level, selectedSectionId, selectedStudentId} = this.props;
+    return onClick
```

Comment:
Would we ever want an onClick and url? For example the onClick may trigger a firehose event.

Reply:
although i can imagine we might want both at some point, i was simply aiming for feature parity with the previous implementation, which at old line 103 avoids the creation of the url if we have an onClick. i put the comment there to make it clear what we're doing in case at some point we want the ability to use both, but since onClick isn't used by LP and i had to finish this in a hurry, i deemed learning all the ins and outs of how it is used to be out of scope and simply preserved the existing functionality.